src/app/lib/device/airplay.js

```diff
@@ -1,53 +1,80 @@
 (function (App) {
     'use strict';
 
-    var browser = require('airplay-js').createBrowser(),
+    var Bonjour = require('bonjour'),
+        AirPlayProtocol = require('airplay-protocol');
+
+    var bonjourBrowser = new Bonjour(),
         collection = App.Device.Collection;
 
-    var makeID = function (baseID) {
-        return 'airplay-' + baseID.replace(':', '');
+    var makeID = function (bonjourService) {
+        // strong ID since there can only be 1 bonjour service per host+port
+        var uniqueDeviceId = bonjourService.host + '_' + bonjourService.port;
+        return 'airplay-' + Common.md5(uniqueDeviceId);
```

Comment:
better keep the full string, there is no use of having the md5 and it makes it harder to debug